token/client/src/token.rs

```diff
@@ -68,8 +71,8 @@ pub enum TokenError {
     AccountInvalidAssociatedAddress,
     #[error("invalid auxiliary account address")]
     AccountInvalidAuxiliaryAddress,
-    #[error("proof error: {0}")]
-    Proof(ProofError),
+    #[error("proof generation")]
+    ProofGeneration,
```

Comment:
Are there plans for token-client to do more with proofs than just generation? Ie is it also going to decrypt / verify / deserialize and all that? If the answer is "yes", I would lean towards keeping the error as it is, and have functions like `generate_proof_data` functions return `ProofError` instead of `TokenError`. That way you don't need `TokenError::ProofGeneration`. What do you think?

Reply:
Yeah, the client would want to decrypt account data (`TokenError::AccountDecryption`) and generate proofs (`TokenError::ProofGeneration`), but I can't really think of a case where it would verify proofs.

Reply:
Per offline conversation, this approach is ok since the other `ProofError`s are either no longer used or already covered